### ISSUE TITLE
Date Picker: screen reader accessibility issues

### ISSUE THREAD

**Issue by ronak-92** [OfficeDev/office-ui-fabric-react#1584]
### Bug Report

- __Package version(s)__: 2.10.3
- __Browser and OS versions__: edge+firefox 

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes, but I am not sure how to resolve it.

Requested priority: Normal 

Products/sites affected: VSTS

#### Describe the issue:

The date picker doesn't read out appropriate messages to give visually impaired users to use the control effectively. These are following issues.
1. When the focus is on input element of date picker.
2. Tabbing to following elements in calendar control and verify inspect properties
- 'Previous Month' 
- Next Month
- Previous Year
- Next Year
- Months (Jan, Feb to Dec)
- Go to today
3. Tab to a date and verify inspect properties.
4. Verify the inspect properties for the month data grid present in the page.

#### Actual behavior:
1. Help text is null. Narrator reads as  '\<aria-label that we have set>, editing' and does not give any information of how it is invokable or is it invokable.
2. Name property is null. Narrator reads as 'group'.
3. Name property is <number> eg: '28'. Narrator reads as '28'.
4.
- IsGridPatternAvailable and IsTablePatternAvailable is true for the datagrid
- Because of the above properties, Narrator reads the date as 'table , 20' for the first time when user focuses on the date

#### Expected behavior:

1. Help text should be 'Press Enter to open the Calendar to select a date' or any such information so that the user will know that there is a calendar control.
2. Name property should be appropriate, simple and should define the purpose of the element. Name property for month names should be '\<month>,\<year>'.
3. Name property should be '\<date>, \<day>, \<month>, \<year>' eg. 20 Thursday April 2017 
4. IsGridPattern and IsTablePattern properties should be false for the datagrid

**Comment by bharay** [comment]
This issue exits in **Dropdown** as well. 
Name property value for is set to null
Narrator announce "editable combo box, '_the default option_', default"
